pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -317,6 +316,35 @@ func (r *ReconcileClusterDeployment) Reconcile(ctx context.Context, request reco
 		return reconcile.Result{}, nil
 	}
 
+	// Remove legacy conditions if present:
+	legacyConditions := []string{
+		"ClusterImageSetNotFound",
+	}
+	newConditions = []hivev1.ClusterDeploymentCondition{}
+	for _, c := range cd.Status.Conditions {
+		legacy := false
+		for _, lc := range legacyConditions {
+			if string(c.Type) == lc {
+				legacy = true
+				break
+			}
+			if !legacy {
+				newConditions = append(newConditions, c)
+			} else {
+				cdLog.Infof("removing legacy condition: %v", c)
+			}
+		}
+	}
+	if len(newConditions) < len(cd.Status.Conditions) {
```

Comment:
Is this future-proof? Would it be better to use a bool flag set to `true` in the loop above? (Like `legacy`, but wider scoped.)

Reply:
Hmm perhaps if someone injected some code in the middle of these statements. I'll add a bool.